File: solutions/python3/0057.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        index = len(intervals)
        for i in range(len(intervals)):
            if newInterval[0] < intervals[i][0]:
                index = i
                break
        intervals.insert(index, newInterval)

        ret = []
        for interval in intervals:
            if not ret or interval[0] > ret[-1][1]:
                ret.append(interval)
            else:
                ret[-1][1] = max(ret[-1][1], interval[1])

        return ret

File: solutions/cpp/0057.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        auto it = intervals.begin();
        while (it != intervals.end() && (*it)[0] < newInterval[0]) it++;
        intervals.insert(it, newInterval);

        vector<vector<int>> ans;
        for (const auto& interval : intervals) {
            if (ans.empty() || interval[0] > ans.back()[1])
                ans.push_back(interval);
            else
                ans.back()[1] = max(ans.back()[1], interval[1]);
        }

        return ans;
    }
};